### PR TITLE
[SPARK-47006][CORE] Refactor `refill()` method to `isExhausted()` in `NioBufferedFileInputStream`

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
The proposed changes in this pr involve refactoring the `refill()` method in the `NioBufferedFileInputStream` class to `isExhausted()`. All calls to `!refill()` within the class have been updated to call `isExhausted()` instead.

### Why are the changes needed?
The changes are needed to improve the readability and understandability of the code. Currently, the `refill()` method in `NioBufferedFileInputStream` is always invoked in a negated context (`!refill()`), which can be confusing and counter-intuitive. 

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GitHub Actions

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

**Comment by LuciferYang** [comment]
Test first

**Review by mridulm — COMMENTED** [review]
IMO `refill` makes more sense here, and conveys the nature of side-effect of invoking the method.
`isExhausted` reads like a getter, but has side effect.

File: core/src/main/java/org/apache/spark/io/NioBufferedFileInputStream.java [hunk 1]
```diff
@@ -66,32 +66,30 @@ private boolean refill() throws IOException {
...
+      return nRead >= 0;
     }
     return true;
   }
 
   @Override
   public synchronized int read() throws IOException {
-    if (!refill()) {
-      return -1;
+    if (refill()) {
```
Comment (LuciferYang) [inline core/src/main/java/org/apache/spark/io/NioBufferedFileInputStream.java:76]: @mridulm I agree with your viewpoint. Currently, I have transformed the code from checking `!refill()` to checking `refill()`, and adjusted the corresponding handling logic (I have not yet updated the PR description). Do you consider this to be acceptable? If not, I will close this PR.

**Comment by LuciferYang** [comment]
Close this one

**Pull request closed without merging**